clients/go/marshal.go

```diff
@@ -0,0 +1,37 @@
+package raccoon
+
+import (
+	"errors"
+
+	"encoding/json"
+
+	"google.golang.org/protobuf/proto"
+)
+
+type Message interface{}
+
+var (
+	// json clickstream marshaler
+	JSON = json.Marshal
+
+	// proto clickstream marshaler
+	PROTO = func(m interface{}) ([]byte, error) {
+		msg, ok := m.(proto.Message)
+		if !ok {
+			return nil, errors.New("unable to marshal non proto")
+		}
+		return proto.Marshal(msg)
+	}
+)
+
+// Marshaler defines a conversion for clickstream message to byte sequence.
+type Marshaler interface {
```

Comment:
@AkbaraliShaikh Any reason to use Marshaler instead of Serialiser?

Reply:
Marshaler and serialiser are synonymous, with the exception that serialisation usually refers to converting an entire object into a byte array.

Marshalling serialises object parameters before adding them to the message and sending it across the network.

I thought the name Marshaler would be appropriate here.

Reply:
I have updated the PR and renamed to `Serializer`, Please review